BnbnavNetClient/Views/MapView.axaml.cs

```diff
@@ -52,20 +60,60 @@ protected override void OnInitialized()
 
             // We need to pan _more_ when scale is smaller:
             MapViewModel.Pan += (_pointerPrevPosition - pointerPos) / MapViewModel.Scale;
+            
+            _pointerVelocities.Add(_pointerPrevPosition - pointerPos);
+
+            if (_pointerVelocities.Count > 5)
+                _pointerVelocities.RemoveAt(0);
+
+            var xAverage = _pointerVelocities.Average(i => i.X);
+            var yAverage = _pointerVelocities.Average(i => i.Y);
+
+            _viewVelocity = new(xAverage, yAverage);
+
+            /*
+             * The actual view velocity should be the average of the last 5
+             * computed velocities, due to how low-quality mouses work (low-quality
+             * mouses have a tendency to move in angles snapped to 45 degrees).
+             */
+
             _pointerPrevPosition = pointerPos;
         };
 
         PointerReleased += (_, __) =>
         {
             _pointerPressing = false;
+
+            _pointerVelocities.Clear(); // Make sure we're not using velocities from previous pan.
         };
 
         PointerWheelChanged += (_, eventArgs) =>
         {
             var deltaScale = eventArgs.Delta.Y * MapViewModel.Scale / 10.0;
             Zoom(deltaScale, (eventArgs.GetPosition(this)));
+
+            _viewVelocity = Vector.Zero; // Reset velocities
+            _pointerVelocities.Clear();
         };
 
+        // This is the physics part of inertial panning.
+        Clock = new Clock();
+        Clock.Subscribe(
+            ts => {
+                if (_pointerPressing)
+                    return;
+
+                // Stop the timer, don't waste resources.
+                if (Math.Abs(_viewVelocity.X) < 4 && Math.Abs(_viewVelocity.Y) < 4) {
+                    _viewVelocity = Vector.Zero;
+                } else {
+                    MapViewModel.Pan += _viewVelocity / MapViewModel.Scale;
+                }
+
+                _viewVelocity /= 1.075; // 1.075 is the friction.
```

Comment:
Newline braces please

Reply:
also, if you're just going to have one statement, just don't do braces at all IMO
(which is what you already did above)